extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote-init.ts

```diff
@@ -333,9 +334,29 @@ class PluginDeployerHandlerImpl implements PluginDeployerHandler {
     // announced ?
     private announced = false;
 
-    constructor(
-        @inject(HostedPluginReader) private readonly reader: HostedPluginReader,
-    ) {
+    /**
+     * Managed plugin metadata backend entries.
+     */
+    private readonly deployedBackendPlugins = new Map<string, DeployedPlugin>();
+
+    @inject(HostedPluginReader)
+    private readonly reader: HostedPluginReader;
+
+    private backendPluginsMetadataDeferred = new Deferred<void>();
+
+    async getDeployedFrontendPluginIds(): Promise<string[]> {
+        return [];
+    }
+
+    async getDeployedBackendPluginIds(): Promise<string[]> {
+        // await first deploy
+        await this.backendPluginsMetadataDeferred.promise;
+        // fetch the last deployed state
+        return [...this.deployedBackendPlugins.keys()];
+    }
+
+    getDeployedPlugin(pluginId: string): DeployedPlugin | undefined {
+        return this.deployedBackendPlugins.get(pluginId);
```

Comment:
Maybe for this method we need `await this.backendPluginsMetadataDeferred.promise;`  too and then return?

Reply:
I don't see why we would need await ?

it's a copy of 
https://github.com/eclipse-theia/theia/blob/6347c5ccbf016b7cfa70976f69616eb1bb49c5b0/packages/plugin-ext/src/hosted/node/hosted-plugin-deployer-handler.ts#L60-L62
and there is not